### PR TITLE
doc: Add Bitbucket bad application id troubleshooting [CY-985]

### PR DESCRIPTION
**Pull request by rtfpessoa**

### CODE REVIEW

File: docs/configuration/integrations/bitbucket-server.md [hunk 1]
```diff
@@ -26,3 +26,15 @@ Follow the instructions below to set up the Codacy integration with Bitbucket Se
...
 After this is done you will be able to use Bitbucket Server to authenticate to Codacy.
+
+## Troubleshooting
+
+- During an authentication procedure, if you got stuck on the provider and this message is shown
+  - ![Invalid client id](./bitbucket-invalid-client-id.png)
+    It means you have not introduced the client id when configuring Bitbucket.com on Codacy.
+    - Make sure the value matches the one in your Bitbucket.com application
+    - If you still could not find the problem:
+        - Extract the parameter `client_id`, from the browser address bar in Bitbucket.com where the error appears (e.g.: `r8QJDkkxj8unYfg4Bd`)
```
Comment (jllopes) [inline docs/configuration/integrations/bitbucket-server.md:37]:
```suggestion
        - Extract the parameter `client_id`, from the browser address bar in your Bitbucket Server URL where the error appears (e.g.: `r8QJDkkxj8unYfg4Bd`)
```

File: docs/configuration/integrations/bitbucket-server.md [hunk 2]
```diff
@@ -26,3 +26,15 @@ Follow the instructions below to set up the Codacy integration with Bitbucket Se
     ```
 
 After this is done you will be able to use Bitbucket Server to authenticate to Codacy.
+
+## Troubleshooting
+
+- During an authentication procedure, if you got stuck on the provider and this message is shown
+  - ![Invalid client id](./bitbucket-invalid-client-id.png)
+    It means you have not introduced the client id when configuring Bitbucket.com on Codacy.
+    - Make sure the value matches the one in your Bitbucket.com application
```
Comment (jllopes) [inline docs/configuration/integrations/bitbucket-server.md:35]:
```suggestion
    - Make sure the value matches the one in your Bitbucket Server application
```

File: docs/configuration/integrations/bitbucket-server.md [hunk 3]
```diff
@@ -26,3 +26,15 @@ Follow the instructions below to set up the Codacy integration with Bitbucket Se
...
+    - If you still could not find the problem:
+        - Extract the parameter `client_id`, from the browser address bar in Bitbucket.com where the error appears (e.g.: `r8QJDkkxj8unYfg4Bd`)
+        - Check if the application id on Bitbucket.com matches this value
```
Comment (jllopes) [inline docs/configuration/integrations/bitbucket-server.md:38]:
```suggestion
        - Check if the application id on Bitbucket Server matches this value
```

Comment (rtfpessoa) [reply docs/configuration/integrations/bitbucket-server.md:38]: BB Server was not supposed to be committed. All the errors will be handled in our side.

File: docs/configuration/integrations/bitbucket-server.md (hunk 2 above)
Comment (rtfpessoa) [reply docs/configuration/integrations/bitbucket-server.md:35]: BB Server was not supposed to be committed. All the errors will be handled in our side.

File: docs/configuration/integrations/bitbucket-server.md (hunk 1 above)
Comment (rtfpessoa) [reply docs/configuration/integrations/bitbucket-server.md:37]: BB Server was not supposed to be committed. All the errors will be handled in our side.

**Comment by rtfpessoa** [comment]
yup, all errors should be workable on our side.

**Comment by rtfpessoa** [comment]
@jllopes @pauloribeiro-codacy fixed the conflicts

**Pull request merged**